types/invoker.go

```diff
@@ -19,6 +20,7 @@ type Invoker struct {
 }
 
 type InvokerResponse struct {
+	Context  context.Context
```

Comment:
Should the context really be sent back with the Response?  How would that be used? Do you envision it behaving like the `http.Response` object with its own `r.Context()`?  I wonder if that is actually recommended, I think the `http.Response` object has an embedded context because they didn't want to change the `http.Handler` interface

Reply:
Hi @LucasRoesler, thank you for reviewing.

> Should the context really be sent back with the Response?

I believe so.

> How would that be used?

I've provided an example [above](https://github.com/openfaas-incubator/connector-sdk/pull/25#issuecomment-519025816).

> Do you envision it behaving like the http.Response object with its own r.Context()?

I don't think I quite understand what you are referring to, but the way I envision this context to work is as a something that groups together contextual information that can be used to correlate responses with the original request as detailed above.

Reply:
I have thought more about this and I don't see an alternative approach.  Adding the context here seems to make sense to me

Also, I realize that I have a typo in my original comment, I meant `http.Request`, not `http.Response`.

Reply:
A custom string, or a map could be used instead of the context. 

My fear is that people will expect the context to be used as something more than the intent Bruno is proposing (as a map/bag).

The context object also has timeout / duration / cancellation / done behaviour, but this behaviour hasn't been implemented in the PR.

Would a map of `string[string]` be sufficient for what you need Bruno, or a string of some sort?

Reply:
Lucas has just pointed me at this line -> https://github.com/openfaas-incubator/connector-sdk/pull/25/files#diff-cd0839c41d9c260fb20a0c53fe284e3eR86

If this context is being used for more than just a bag for a request / message ID and is fully functional as a Context, I think Lucas + I are in agreement with moving the change forward with #28 raised to cover my other concerns.